### DISCUSSION TITLE
Unclear on some things

### DISCUSSION THREAD

**Discussion by minus7** [pgexporter/pgexporter#3] [General]
Hi, reading the readme and docs leaves me somewhat unclear on some things:

- What data does pgexporter export? Can i monitor replication lag with it? A list or sample would be nice in the readme
- How to add it to Prometheus? Is any secret/user necessary? What's the URL?
- How does pgexporter interact with Postgres? the readme mentions "process model" and "shared memory" but I think that refers to pgexporter itself? Only in the configuration documentation it becomes apparent that pgexporter connects to Postgres as a user with special privileges.

I think half of that could be addressed in the introductory sentence, e.g.: `pgexporter` is a Prometheus exporter that remotely connects to a PostgreSQL server and lets you monitor replication status, throughput, latencies and much more. (replace with a list of prominent features that actually apply)

**Comment by jesperpedersen** [comment]
Hi,

- I have added https://pgexporter.github.io/metrics.html to list the current metrics. Currently `pg_replication_slots` only has `active`, but lag is def something that is needed. Feel free to file feature requests for them - adding a new metrics would be done in `queries.h`, `queries.c` and `prometheus.c` - just follow the implementation of an existing metrics.
- Follow https://pgexporter.github.io/gettingstarted.html and that will give you an `localhost:5002` endpoint (you can test it with Firefox, curl, ...). Then you point - f.ex. Grafana - to it as a Prometheus datasource.
- pgexporter connects to PostgreSQL using the defined user in the configuration - the rest is really internals. If there is something unclear in the Getting Started guide file an issue for it, please.

You can deploy pgexporter local (same machine as PostgreSQL), or as a remote service (different machine than PostgreSQL). You can monitor one or more PostgreSQL instances with a single pgexporter deployment. It really depends on your PostgreSQL architecture.

Hope this helps !

**Reply by minus7** [reply]
Thanks!

**Comment by marcbachmann** [comment]
And what's the motivation behind the project?
How is it different than https://github.com/prometheus-community/postgres_exporter or https://vector.dev/docs/reference/configuration/sources/postgresql_metrics/, which already provide a quite good feature set?

**Reply by jesperpedersen** [reply]
Its an alternative implementation. C based with a 3-Clause BSD license, and packages available in the PostgreSQL community YUM repository.

Features will be similar, but there will be different ones as well - the project is community driven so what ever features people are looking for are likely to get in. Of course, we have internal requirements at Red Hat as well that I'll be working on.

As always in the open source world - find the solution that fits your architecture the best, and help improve the project in question.
